WordPress/gutenberg
# Mobile: BottomSheet design tweaks v2

## Description

This PR improves the design of the BottomSheet following https://github.com/wordpress-mobile/gutenberg-mobile/issues/535 .
Also added fixed for:
https://github.com/wordpress-mobile/gutenberg-mobile/issues/597
https://github.com/wordpress-mobile/gutenberg-mobile/issues/596

- Added more space at the top of the BottomSheet (it should sum up to 24)
- Make last row’s border-bottom full-width.
  - Added separator styles: `fullWidth`, `leftMargin`, and `none`. If it's not specified, it will behave as before.
  - @marecar3 - This change might hit the BottomSheet on your PR too.
- Truncate long text strings in the middle instead of at the end of the line.
  - This one was tricky since TextInput doesn't have this configuration. I ended up replacing the TextInput with a Text component to display the value, and change it back to TextInput when the user want's to edit it.
- Force title case for the row titles.
  - We don't have a "native" way of doing it in RN (not supported), so I opted for doing it manually for now.

![bottom-sheet-design](https://user-images.githubusercontent.com/9772967/52711705-c8141200-2f92-11e9-9831-6f1e247d2c7e.png)

**To test:**
- Check out the related `gutenberg-mobile` PR (https://github.com/wordpress-mobile/gutenberg-mobile/pull/594).
- Display the Image Settings BottomSheet.
- Check that the placeholders are displaying properly.
- Check that you can edit the settings by pressing on the cells.
- Check that the design looks good 😎 

Swipe:
- Test swipe to dismiss from anywhere in the BottomSheet.
